src/devices/obs.ts

```diff
@@ -0,0 +1,740 @@
+
+		let oldStreaming = oldState.streaming
+		let newStreaming = newState.streaming
+		if (newStreaming !== undefined) {
+			if (oldStreaming !== newStreaming) {
+				commands.push({
+					command: {
+						requestName: newStreaming
+							? OBSRequestName.START_STREAMING
+							: OBSRequestName.STOP_STREAMING,
+						args: {}
+					},
+					context: 'streaming changed',
+					timelineId: ''
+				})
+			}
+		}
+
+		return commands
+	}
+
+	private _resolveMute (
+		oldState: OBSState,
+		newState: OBSState
+	): Array<OBSCommandWithContext> {
+		let commands: Array<OBSCommandWithContext> = []
+
+		let oldMuted = oldState.muted
+		let newMuted = newState.muted
+		Object.keys(newMuted).forEach((source) => {
+			if (
+				newMuted[source] !== oldMuted[source]
+			) {
+				commands.push({
+					command: {
+						requestName: OBSRequestName.SET_MUTE,
+						args: {
+							source: source,
+							mute: newMuted[source]
+						}
+					},
+					context: `mute ${source} changed`,
+					timelineId: ''
+				})
+			}
+		})
+
+		return commands
+	}
+
+	private _resolveScenes (
+		oldState: OBSState,
+		newState: OBSState
+	): Array<OBSCommandWithContext> {
+		let commands: Array<OBSCommandWithContext> = []
+
+		let oldScenes = oldState.scenes
+		let newScenes = newState.scenes
+		Object.entries(newScenes).forEach(([sceneName, scene]) => {
+			Object.entries(scene.sceneItems).forEach(
+				([source, newSceneItemProperties]) => {
+					let oldSceneItemProperties =
+						oldScenes[sceneName]?.sceneItems[source]
+					if (
+						oldSceneItemProperties as any === undefined ||
+							newSceneItemProperties.render !==
+							oldSceneItemProperties.render
+					) {
+						commands.push({
+							command: {
+								requestName: OBSRequestName.SET_SCENE_ITEM_RENDEER,
+								args: {
+									'scene-name': sceneName,
+									source: source,
+									render: newSceneItemProperties.render
+								}
+							},
+							context: `scene ${sceneName} item ${source} changed render`,
+							timelineId: ''
+						})
+					}
+				}
+			)
+		})
+
+		return commands
+	}
+
+	private _resolveSourceSettings (
+		oldState: OBSState,
+		newState: OBSState
+	): Array<OBSCommandWithContext> {
+		let commands: Array<OBSCommandWithContext> = []
+
+		let oldSources = oldState.sources
+		let newSources = newState.sources
+		Object.entries(newSources).forEach(([sourceName, source]) => {
+			if (!_.isEqual(source.sourceSettings, oldSources[sourceName]?.sourceSettings)) {
+				commands.push({
+					command: {
+						requestName: OBSRequestName.SET_SOURCE_SETTINGS,
+						args: {
+							sourceName: sourceName,
+							sourceSettings: source.sourceSettings
+						}
+					},
+					context: `source ${sourceName} changed settings`,
+					timelineId: ''
+				})
+			}
+		})
+
+		return commands
+	}
+
+	private _diffStates (
+		oldState: OBSState,
+		newState: OBSState
+	): Array<OBSCommandWithContext> {
+		let commands: Array<OBSCommandWithContext> = []
+
+		commands = commands.concat(
```

Comment:
Feel free to ignore this style preference

I think it is nicer to either use `commands.push(...this._resolveThing())` (so that commands can be const) or use the spread operator while building the array
```
const commands = [
...this._resolveThing(),
...this._resolveThing(),
]
```